doc/source/reference/index.rst

```diff
@@ -24,13 +24,14 @@ The following subpackages are public.
   `pandas-stubs <https://github.com/pandas-dev/pandas-stubs>`_ package
   which has classes in addition to those that occur in pandas for type-hinting.
 
-In addition, public functions in ``pandas.io`` and ``pandas.tseries`` submodules
-are mentioned in the documentation.
+In addition, public functions in ``pandas.io``, ``pandas.tseries``, ``pandas.util`` submodules
+are explicitly mentioned in the documentation. Further APIs in these modules are not guaranteed
+to be stable.
 
 
 .. warning::
 
-    The ``pandas.core``, ``pandas.compat``, and ``pandas.util`` top-level modules are PRIVATE. Stable functionality in such modules is not guaranteed.
+    The ``pandas.core``, ``pandas.compat`` top-level modules are PRIVATE. Stable functionality in such modules is not guaranteed.
```

Comment:
I've probably just missed this conversation along the way but what prevents us from prefixing these from an underscore to more clearly dictate that intent?

Reply:
Especially for `pandas.core`, I think there's an unknown amount of use of internals and I think a conservative approach is desired (i.e. formally deprecate) to wean users off of it https://github.com/pandas-dev/pandas/issues/27522. Most pertinent in my mind is objects that can be returned that are not exposed in the public API https://github.com/pandas-dev/pandas/issues/55905